clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -3652,17 +3657,23 @@ scheduler_tmpl_t<RecordType, ReaderType>::next_record(output_ordinal_t output,
     // do return an error on a time smaller than an input's current start time when we
     // check for quantum end.
     if (cur_time == 0) {
-        // It's more efficient for QUANTUM_INSTRUCTIONS to get the time here instead of
-        // in get_output_time().  This also makes the two more similarly behaved with
-        // respect to blocking system calls.
-        // TODO i#6971: Use INSTRS_PER_US to replace .cur_time completely
-        // with a counter-based time, weighted appropriately for STATUS_IDLE.
-        cur_time = get_time_micros();
+        if (options_.mapping == MAP_AS_PREVIOUSLY) {
+            // XXX: Should we instead store the replay time (whether passed in or our
+            // instr-based formula below) in the records and do away with wall-clock
+            // time altogether?  Either way, we should make it clear in the docs
+            // whether the user/simulator has to pass in the time on replay.
```

Comment:
I filed #7023 on this.  Wall-clock time for coordinating concurrent outputs on replay will remain no matter what the simulation time does.  But currently replay is using wall clock to compute idle durations.  Changing this may require a replay file format change.  Separated it into #7023.